integrations/sharepoint-excel/src/actions/sync-excel-file.ts

```diff
@@ -0,0 +1,588 @@
+  const getRealClient = (client: sdk.IntegrationSpecificClient<bp.TIntegration>) => (client as any)._client
+  const realClient = getRealClient(client)
+
+  const { sharepointFileUrl, sheetTableMapping } = input
+  logger.forBot().info(`Syncing Excel file: "${sharepointFileUrl}"`)
+  logger.forBot().info(`Using sheetTableMapping: ${sheetTableMapping}`)
+
+  const spClient = new SharepointClient({
+    primaryDomain: ctx.configuration.primaryDomain,
+    siteName: ctx.configuration.siteName,
+    clientId: ctx.configuration.clientId,
+    tenantId: ctx.configuration.tenantId,
+    thumbprint: ctx.configuration.thumbprint,
+    privateKey: ctx.configuration.privateKey,
+  })
+
+  try {
+    logger.forBot().debug(`Fetching Excel file from URL: ${sharepointFileUrl}`)
+
+    const fileContentBuffer = await fetchExcelFile(spClient, sharepointFileUrl, ctx.configuration.siteName, logger)
+
+    const workbook = xlsx.read(fileContentBuffer, { type: 'buffer' })
+    logger
+      .forBot()
+      .info(`Excel workbook loaded with ${workbook.SheetNames.length} sheet(s): ${workbook.SheetNames.join(', ')}`)
+
+    // Determine which sheets to process
+    let sheetsToProcess: string[] = []
+    let sheetToTable: Record<string, string> = {}
+
+    // Parse sheetTableMapping
+    try {
+      if (sheetTableMapping.trim().startsWith('{')) {
+        sheetToTable = JSON.parse(sheetTableMapping)
+      } else {
+        // Parse as comma-separated pairs: Sheet1:table1,Sheet2:table2
+        sheetTableMapping.split(',').forEach((pair: string) => {
+          const [sheet, table] = pair.split(':').map((s: string) => s.trim())
+          if (sheet && table) sheetToTable[sheet] = table
+        })
+      }
+      sheetsToProcess = Object.keys(sheetToTable)
+      logger.forBot().info(`Parsed sheetTableMapping: ${JSON.stringify(sheetToTable)}`)
+    } catch (err) {
+      logger.forBot().error(`Failed to parse sheetTableMapping: ${err}`)
+      throw new sdk.RuntimeError('Invalid sheetTableMapping format. Use JSON or comma-separated pairs.')
+    }
+
+    // Validate that all sheets in the mapping exist in the workbook
+    const missingSheets = sheetsToProcess.filter((sheet) => !workbook.SheetNames.includes(sheet))
+    if (missingSheets.length > 0) {
+      throw new sdk.RuntimeError(
+        `Sheets not found in workbook: ${missingSheets.join(', ')}. Available sheets: ${workbook.SheetNames.join(', ')}`
+      )
+    }
+
+    const processedSheets: any[] = []
+    const failedSheets: any[] = []
+
+    // Process each sheet
+    for (const currentSheetName of sheetsToProcess) {
+      logger.forBot().info(`\n--- Processing sheet: "${currentSheetName}" ---`)
+
+      const worksheet = workbook.Sheets[currentSheetName]
+      if (!worksheet) {
+        logger.forBot().warn(`Sheet "${currentSheetName}" is undefined, skipping`)
+        continue
+      }
+
+      const jsonData = xlsx.utils.sheet_to_json(worksheet, { header: 1 })
```

Comment:
todo: rename this to jsonArray or something indicating it's a list. this is not intuitive enough.